x-pack/test/functional/page_objects/lens_page.ts

```diff
@@ -287,6 +293,22 @@ export function LensPageProvider({ getService, getPageObjects }: FtrProviderCont
         await testSubjects.setEuiSwitch('returnToOriginModeSwitch', state);
       }
 
+      const dashboardSelectorExists = await testSubjects.exists('add-to-dashboard-options');
+      if (dashboardSelectorExists) {
+        const dashboardSelector = await testSubjects.find('add-to-dashboard-options');
+        let optionSelector = 'add-to-library-option';
+        if (addToDashboard) {
+          optionSelector = dashboardId ? 'existing-dashboard-option' : 'new-dashboard-option';
+        }
+        log.debug('dashboard selector exists, choosing option:', optionSelector);
+        const label = await dashboardSelector.findByCssSelector(`label[for="${optionSelector}"]`);
+        await label.click();
+
+        if (dashboardId) {
+          // TODO - selecting an existing dashboard
```

Comment:
I'm a bit confused with the flow here: is the goal to ensure just that the all dashboard selectors exist, without actually adding the Lens visualization to the dashboard?

Reply:
as part of https://github.com/elastic/kibana/pull/83140, a new selector was added to the visualize save modal to allow users to add by value directly to a dashboard from visualize. That also means that in order to save it to the library, the user needs to select the `add to library` option on the dashboard selector.

What I've added here is a default path that says 'if the dashboard selection section exists, and we haven't been asked to add it directly to a dashboard, select the `add to library` option. This prevents a number of test failures - and you can see a similar implementation in the Visualize page.

Reply:
> What I've added here is a default path that says 'if the dashboard selection section exists, and we haven't been asked to add it directly to a dashboard, select the add to library option. This prevents a number of test failures - and you can see a similar implementation in the Visualize page.

nit: It might be helpful to extract this logic into another helper function that could be pulled into the lens & SO tagging functional tests. It feels like the type of implementation detail that other apps shouldn't necessarily need to concern themselves with when writing tests.

e.g. something like this could potentially be used everywhere:
```js
await saveToNewDashboard();
await saveToExistingDashboard('my-dashboard-id');
await saveToLibrary();
```